src/hooks/useRegister.ts

```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+import { FirebaseError } from "firebase/app";
+import { doc, setDoc } from "firebase/firestore";
+
+import { createUserWithEmailAndPassword, getAuth } from "firebase/auth";
+
+import { db, firebaseApp } from "./../config/firebase";
+
+type UserData = {
+	firstName: string;
+	lastName: string;
+	email: string;
+	password: string;
+};
+
+export const useRegister = () => {
+	const [error, setError] = useState<string | null>(null);
+	const [isPending, setIsPending] = useState<boolean>(false);
+
+	const auth = getAuth(firebaseApp);
+	const registerUser = async (data: UserData) => {
+		setIsPending(true);
+		setError(null);
+		try {
+			const res = await createUserWithEmailAndPassword(
+				auth,
+				data.email,
+				data.password,
+			);
+			if (!res) throw new Error("Some error during register.");
+
+			const userRef = doc(db, "User", res.user.uid);
+
+			setDoc(userRef, {
+				name: data.firstName,
+				lastName: data.lastName,
+				displayName: `${data.firstName} ${data.lastName}`,
```

Comment:
displayName nie trzymaj w firebase. Duplikujesz dane z które już tam masz i będziesz musiała zawsze je synchronizować po edycji. Jak chcesz mieć takie pole w aplikacji to stwórz je już w kodzie po pobraniu z firebase